### PR TITLE
aws-auth 2.1.4 (new formula)

### PR DESCRIPTION
**Pull request by DocLM**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
upstream missing the license file in the repo which caused the audit failure.

```
$ gh repo view iamarkadyt/aws-auth --json licenseInfo
{
  "licenseInfo": null
}
```

**Comment by DocLM** [comment]
Hi @chenrui333,
I'm working with the maintainer to get this fixed.

File: Formula/aws-auth.rb
Comment (SMillerDev) [inline Formula/aws-auth.rb:17]: Is there no way to only install the native binaries?

Comment (DocLM) [reply Formula/aws-auth.rb]: I removed the lines since there are no native binaries

**Comment by DocLM** [comment]
License issue should be fixed now

File: Formula/aws-auth.rb
Comment (SMillerDev) [inline Formula/aws-auth.rb:19]: No need to check if brew works.

File: Formula/aws-auth.rb
Comment (SMillerDev) [inline Formula/aws-auth.rb:24]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Comment (iamarkadyt) [reply Formula/aws-auth.rb:24]: Maintainer here. This library does need credentials to operate. It has a dependency on user-added AWS accounts that it hooks into. Some ideas for tests:
* Test `aws-auth config` command, and that a simple configuration can be added properly.
* Test that `aws-auth login` invokes an interactive prompt asking to decrypt the configuration file, fails with an error when configuration is missing, or fails if the password is incorrect.
* Test `aws-auth reset` command that erases the configuration file.

**Comment by chenrui333** [comment]
@DocLM I think we can update to the 2.1.4 release, and make the tests a bit better, it should be good to go after that. Thanks for the work!

**Comment by DocLM** [comment]
@chenrui333 Now should be on the latest version

File: Formula/aws-auth.rb
Comment (DocLM) [reply Formula/aws-auth.rb:24]: Hi @SMillerDev, as @iamarkadyt all the commands require interactivity (also reset).
I'm actually trying to test `aws-auth login` without luck, at the moment I did not find a way to use `shell_output` or `pipe_output` with that kind of interactive prompt.

Comment (DocLM) [reply Formula/aws-auth.rb]: I've updated the test using pty and simulating an interactive session with `aws-auth login`.
@SMillerDev do you think that it's enought?

Comment (SMillerDev) [reply Formula/aws-auth.rb]: Yeah, this test makes sense to me, thanks @DocLM !

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
LGTM, thanks a lot for the work @DocLM !

**Pull request closed without merging**
